cli_tools/gce_ovf_import/main.go

```diff
@@ -0,0 +1,501 @@
+	translateWorkflowPath string,
+	bootDiskGcsPath string,
+	machineType string) map[string]string {
+	varMap := map[string]string{}
+
+	varMap["instance_name"] = strings.ToLower(*instanceNames)
+	if translateWorkflowPath != "" {
+		varMap["translate_workflow"] = translateWorkflowPath
+		varMap["install_gce_packages"] = strconv.FormatBool(!*noGuestEnvironment)
+	}
+	if bootDiskGcsPath != "" {
+		varMap["boot_disk_file"] = bootDiskGcsPath
+	}
+	if *network != "" {
+		varMap["network"] = fmt.Sprintf("global/networks/%v", *network)
+	}
+	if *subnet != "" {
+		varMap["subnet"] = fmt.Sprintf("regions/%v/subnetworks/%v", *region, *subnet)
+	}
+	if machineType != "" {
+		varMap["machine_type"] = machineType
+	}
+	if *description != "" {
+		varMap["description"] = *description
+	}
+	if *privateNetworkIP != "" {
+		varMap["private_network_ip"] = *privateNetworkIP
+	}
+
+	if *networkTier != "" {
+		varMap["network_tier"] = *networkTier
+	}
+	//TODO: networkInterface
+	return varMap
+}
+
+// Daisy workflow files don't support boolean variables so we have to do it here.
+func updateInstanceWithBooleanFlagValues(w *daisy.Workflow) {
+	instance := (*w.Steps["create-instance"].CreateInstances)[0]
+	instance.CanIpForward = *canIPForward
+	instance.DeletionProtection = *deletionProtection
+	if instance.Scheduling == nil {
+		instance.Scheduling = &compute.Scheduling{}
+	}
+	if *noRestartOnFailure {
+		vFalse := false
+		instance.Scheduling.AutomaticRestart = &vFalse
+	}
+}
+
+func toWorkingDir(dir string) string {
+	wd, err := filepath.Abs(filepath.Dir(*currentExecutablePath))
+	if err == nil {
+		return path.Join(wd, dir)
+	}
+	return dir
+}
+
+// creates a new Daisy Compute client
+func createComputeClient(ctx *context.Context) (daisycompute.Client, error) {
+	computeOptions := []option.ClientOption{option.WithCredentialsFile(*oauth)}
+	if *ce != "" {
+		computeOptions = append(computeOptions, option.WithEndpoint(*ce))
+	}
+
+	computeClient, err := daisycompute.NewClient(*ctx, computeOptions...)
+	if err != nil {
+		return nil, err
+	}
+	return computeClient, nil
+}
+
+// OVFImporter is responsible for importing OVF into GCE
+type OVFImporter struct {
+	ctx                   context.Context
+	storageClient         commondomain.StorageClientInterface
+	computeClient         daisycompute.Client
+	tarGcsExtractor       commondomain.TarGcsExtractorInterface
+	mgce                  commondomain.MetadataGCEInterface
+	ovfDescriptorLoader   domain.OvfDescriptorLoaderInterface
+	bucketIteratorCreator commondomain.BucketIteratorCreatorInterface
+	logger                logging.LoggerInterface
+	zoneValidator         commondomain.ZoneValidatorInterface
+	gcsPathToClean        string
+	workflowPath          string
+	buildID               string
+	diskInfos             *[]ovfutils.DiskInfo
+}
+
+// NewOVFImporter creates an OVF importer, including automatically poulating dependencies,
+// such as compute/storage clients.
+func NewOVFImporter() (*OVFImporter, error) {
```

Comment:
Is there a reason this is a public function in main?

Reply:
It can be private. I was thinking of extracting this to a separate package but decided against it.

Reply:
Made private